### PR TITLE
Add -I option to sol-fbp-generator

### PR DESCRIPTION
**Pull request by lgywata**
Now generator is able to search for fbp files according to paths
passed to it through -I option.

Signed-off-by: Luiz Ywata <luizg.ywata@intel.com>

### CODE REVIEW

File: src/bin/sol-fbp-generator/main.c [hunk 1]
```diff
@@ -952,12 +952,57 @@ handle_json_path(const char *path)
...
+handle_include_dir(char *fullpath, const char *basename)
+{
+    struct stat s;
+    const char *p;
+    uint16_t i;
+    int err;
+
+    SOL_PTR_VECTOR_FOREACH_IDX (&args.include_paths, p, i) {
+        err = snprintf(fullpath, PATH_MAX, "%s/%s", p, basename);
+        SOL_INT_CHECK(err, < 0, false);
```
Comment (ricardotk) [inline src/bin/sol-fbp-generator/main.c]: You should check if the returned value is > fullpath as well.
e.g. `if (len < 0 || len > (int)sizeof(fullpath))`

File: src/bin/sol-fbp-generator/main.c [hunk 2]
```diff
@@ -952,12 +952,57 @@ handle_json_path(const char *path)
...
+    struct stat s;
+    char *dup_path;
+
+    if (stat(path, &s) != 0)
+        return false;
+
+    if (S_ISDIR(s.st_mode))
+        dup_path = sol_arena_strdup(str_arena, path);
+    else if (S_ISREG(s.st_mode))
+        dup_path = sol_arena_strdup(str_arena, dirname(path));
```
Comment (ricardotk) [inline src/bin/sol-fbp-generator/main.c]: check if dup_path is `!= NULL`

**Comment by cmarcelo** [comment]
+1 with comments.

**Comment by lgywata** [comment]
Ok, thanks. Applied.

**Pull request closed without merging**